Bottom: 72acdd905531cc01e04ab147ecab0a3b19f3f395
Top:    e7fa6112bb86388735b72f41789d59401c5c4695
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 16:16:57 +0100

Use motion hints when processing pointer motion over the drawing area


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index bad95f9..89b8e38 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -751,7 +751,7 @@ ghid_show_crosshair (gboolean show)
 
   glBegin (GL_LINES);
 
-#if 1
+#if 0
   if (x_prev >= 0)
     draw_crosshair (x_prev, y_prev);
 #endif
@@ -764,7 +764,7 @@ ghid_show_crosshair (gboolean show)
 
   glBegin (GL_QUADS);
 
-#if 1
+#if 0
   if (x_prev >= 0 && draw_markers_prev)
     {
       glVertex2i (0,                  y_prev - VCD);
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index cd21ef3..bf3021e 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -682,12 +682,14 @@ queue_tooltip_update (GHidPort *out)
 
 gint
 ghid_port_window_motion_cb (GtkWidget * widget,
-			    GdkEventButton * ev, GHidPort * out)
+			    GdkEventMotion * ev, GHidPort * out)
 {
   gdouble dx, dy;
   static gint x_prev = -1, y_prev = -1;
   gboolean moved;
 
+  gdk_event_request_motions (ev);
+
 #if 0
   if (!ghid_start_drawing (out))
     return FALSE;
@@ -696,7 +698,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   if (out->panning)
     {
 /* Not sure this really matters, since we're using invalidate to get a redraw */
-#if 1
+#if 0
       if (gtk_events_pending ())
 	return FALSE;
 #endif
@@ -709,7 +711,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
       return FALSE;
     }
   x_prev = y_prev = -1;
-  moved = ghid_note_event_location (ev);
+  moved = ghid_note_event_location ((GdkEventButton *)ev);
 
 #if ENABLE_TOOLTIPS
   queue_tooltip_update (out);
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index bd3ca34..d0e32df 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -2299,7 +2299,8 @@ ghid_build_pcb_top_window (void)
 			 | GDK_LEAVE_NOTIFY_MASK | GDK_ENTER_NOTIFY_MASK
 			 | GDK_BUTTON_RELEASE_MASK | GDK_BUTTON_PRESS_MASK
 			 | GDK_KEY_RELEASE_MASK | GDK_KEY_PRESS_MASK
-			 | GDK_FOCUS_CHANGE_MASK | GDK_POINTER_MOTION_MASK);
+			 | GDK_FOCUS_CHANGE_MASK | GDK_POINTER_MOTION_MASK
+			 | GDK_POINTER_MOTION_HINT_MASK);
 
   /*
    * This is required to get the drawing_area key-press-event.  Also the
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index dec007f..c0d80b8 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -303,7 +303,7 @@ gint ghid_port_window_enter_cb (GtkWidget * widget,
 gint ghid_port_window_leave_cb (GtkWidget * widget, 
                            GdkEventCrossing * ev, GHidPort * out);
 gint ghid_port_window_motion_cb (GtkWidget * widget,
-				 GdkEventButton * ev, GHidPort * out);
+				 GdkEventMotion * ev, GHidPort * out);
 gint ghid_port_window_mouse_scroll_cb (GtkWidget * widget,
 				       GdkEventScroll * ev, GHidPort * out);
